### PR TITLE
Onboarding questions

### LINKED ISSUE #59: Create onboarding questions and seed database
What is your birthday?
When did you graduate HS?
When did you graduate college?
When did you move to your current house?
If you're currently married, what was the date you were married?
What are the approximate dates that you started & ended your first job?
What are the dates of your most memorable vacation?

All of these questions will also have a name/description (see ticket #78)

### LINKED ISSUE #61: US: Get onboarding questions query
Expose the question's ID for the FE

### LINKED ISSUE #78: Add name or description to questions table
Ex: 'When were you married?' could have a name of 'Got married' or something like that. We'll use it to know what to name the user's era/event after onboarding. The FE will use it to display on the calendar.

### PR DESCRIPTION
**Pull request by dhalverson**
### Code Highlights:
- Updates Questions table to include name column
- Adds Question Type
- Adds getOnboardingQuestions Query and Testing (this will return only questions where onboarding is marked as true)

### Where should the reviewer start?
 Spec > Questions> get_questions_spec
 GraphQL > Types > question_type
GraphQL > Types > query_type

### Have Tests Been Added?
- [ ] No.
- [ ] Yes, but all tests are not passing.
- [x] Yes, and all are passing.
### Any background context you want to provide?
NA
### Message/Questions for reviewer:
Tests have been written, will likely go back and add edge cases later. Our test coverage is low at the moment.

### Issues:
* Addresses #
* Closes #78 
* Closes #59 
* Closes #61
### Screenshots (if appropriate):
NA

### Tracking Consistency:
- [x] added appropriate labels
- [x] My code follows the code style of this project and has removed all unnecessary annotations
- [x] I have added comments on my pull request, particularly in hard-to-understand areas
- [x] looked at PR preview to check spelling, syntax, formatting, and completion
- [x] Rubocop Violations:

### CODE REVIEW

File: app/graphql/types/query_type.rb [hunk 1]
```diff
@@ -15,6 +15,10 @@ class QueryType < Types::BaseObject
       argument :id, ID, required: true
     end
 
+    # Questions
+    field :get_onboarding_questions, [Types::QuestionType], null: false,
```
Comment (dhalverson) [inline app/graphql/types/query_type.rb:19]: Will return an array of hashes, which consist of questions where onboarding is marked as true. This allows us to send only onboarding questions.

File: app/models/question.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class Question < ApplicationRecord
```
Comment (dhalverson) [inline app/models/question.rb:1]: Question model with validations

File: db/migrate/20210105214942_add_name_to_questions.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddNameToQuestions < ActiveRecord::Migration[6.1]
```
Comment (dhalverson) [inline db/migrate/20210105214942_add_name_to_questions.rb:1]: Migration file to add name to questions table

File: db/seeds.rb [hunk 4]
```diff
@@ -14,10 +14,8 @@
 Era.reset_pk_sequence
 User.destroy_all
 User.reset_pk_sequence
-
-Event.delete_all
```
Comment (dhalverson) [inline db/seeds.rb:18]: Removed these, as they were duplicative from the destroy all

File: db/seeds.rb [hunk 5]
```diff
@@ -27,3 +25,13 @@
 
 FactoryBot.create_list(:event, 3, user: frosty)
 FactoryBot.create_list(:event, 3, user: santa)
+
+Question.create!(name: 'Birthdate', question: 'When is your birthday?', onboarding: true)
```
Comment (dhalverson) [inline db/seeds.rb:29]: Seeded questions from FE

File: db/seeds.rb [hunk 6]
```diff
@@ -27,3 +25,13 @@
...
+Question.create!(name: 'Graducated from high school', question: 'When did you graduate high school?', onboarding: true)
+Question.create!(name: 'Graduated from college', question: 'When did you graduate college?', onboarding: true)
+Question.create!(name: 'Moved to current house', question: 'What is your birthday?', onboarding: true)
+Question.create!(name: 'Got married', question: "If you're currently married, when were you married?", onboarding: true)
+Question.create!(name: 'First job', question: 'What are the approximate dates that you started and ended your first job?', onboarding: true)
+Question.create!(name: 'Best vacation', question: 'What are the dates of your most memorable vacation?', onboarding: true)
+
+print 'Seeding Successful!'
```
Comment (dhalverson) [inline db/seeds.rb]: Added field to alert user of success

File: spec/requests/graphql/queries/questions/get_questions_spec.rb [hunk 7]
```diff
@@ -0,0 +1,42 @@
+require 'rails_helper'
+
+RSpec.describe Types::QueryType do
+  describe 'onboarding questions' do
+    it 'can query all onboarding questions' do
+
+      Question.create!(name: 'Question 1', question: 'What is your favorite color?', onboarding: true)
```
Comment (dhalverson) [inline spec/requests/graphql/queries/questions/get_questions_spec.rb:7]: Generic questions just for testing.

File: .gitignore [hunk 8]
```diff
@@ -34,3 +34,5 @@
 
 # Ignore the manifest.js
 /app/assets/config/manifest.js
+
+/config/application.rb
```
Comment (AngelaGuardia) [inline .gitignore]: Is this an okay file to ignore?

File: db/seeds.rb [hunk 9]
```diff
@@ -27,3 +25,13 @@
 
 FactoryBot.create_list(:event, 3, user: frosty)
 FactoryBot.create_list(:event, 3, user: santa)
+
+Question.create!(name: 'Birthdate', question: 'When is your birthday?', onboarding: true)
+Question.create!(name: 'Graducated from high school', question: 'When did you graduate high school?', onboarding: true)
```
Comment (AngelaGuardia) [inline db/seeds.rb]:
```suggestion
Question.create!(name: 'Graduated from high school', question: 'When did you graduate high school?', onboarding: true)
```

File: db/seeds.rb (hunk 6 above)
Comment (AngelaGuardia) [reply db/seeds.rb]: Like the "seeding successful" feedback!

**Review by AngelaGuardia — APPROVED** [review]
Well done Dan! Minor change for a spelling mistake.

File: .gitignore (hunk 8 above)
Comment (leahriffell) [reply .gitignore]: Hmm I'm not sure. I don't think we want to ignore it because it is needed to make the API run probably, correct?

**Review by leahriffell — CHANGES_REQUESTED** [review]
I think we will want to remove the addition of application.rb to gitignore file but other than that, good to go!

**Comment by dhalverson** [comment]
> I think we will want to remove the addition of application.rb to gitignore file but other than that, good to go!

Updated, sorry I am not sure why I did that

File: .gitignore (hunk 8 above)
Comment (dhalverson) [reply .gitignore]: Sorry, I'll fix that

**Review by leahriffell — APPROVED** [review]

File: spec/requests/graphql/queries/questions/get_questions_spec.rb [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+        getOnboardingQuestions {
+          id
+          name
+          question
+          onboarding
+        }
+      }
+    GQL
+  end
+end
```
Comment (LHJE) [inline spec/requests/graphql/queries/questions/get_questions_spec.rb:42]: All of this looks terrif!

**Review by LHJE — APPROVED** [review]
Stellar job team!
